test/new-e2e/tests/installer/windows/datadog_installer.go

```diff
@@ -0,0 +1,210 @@
+func (d *DatadogInstaller) InstallExperiment(packageName string, opts ...installer.PackageOption) (string, error) {
+	return d.runCommand("install-experiment", packageName, opts...)
+}
+
+// RemovePackage requests that the Datadog Installer removes a package on the remote host.
+func (d *DatadogInstaller) RemovePackage(packageName string) (string, error) {
+	return d.execute(fmt.Sprintf("remove %s", packageName))
+}
+
+// RemoveExperiment requests that the Datadog Installer removes a package on the remote host.
+func (d *DatadogInstaller) RemoveExperiment(packageName string) (string, error) {
+	return d.execute(fmt.Sprintf("remove-experiment %s", packageName))
+}
+
+// InstallerParams contains the optional parameters for the Datadog Installer Install command
+type InstallerParams struct {
+	installerURL string
+}
+
+// InstallerOption is an optional function parameter type for the Datadog Installer Install command
+type InstallerOption func(*InstallerParams) error
+
+// WithInstallerURL uses a specific URL for the Datadog Installer Install command instead of using the pipeline URL.
+func WithInstallerURL(installerURL string) InstallerOption {
+	return func(params *InstallerParams) error {
+		params.installerURL = installerURL
+		return nil
+	}
+}
+
+// WithInstallerURLFromInstallersJSON uses a specific URL for the Datadog Installer from an installers_v2.json
+// file.
+// bucket: The S3 bucket to look for the installers_v2.json file, i.e. "dd-agent-mstesting"
+// channel: The channel in the bucket, i.e. "stable"
+// version: The artifact version to retrieve, i.e. "7.56.0-installer-0.4.5-1"
+//
+// Example: WithInstallerURLFromInstallersJSON("dd-agent-mstesting", "stable", "7.56.0-installer-0.4.5-1")
+// will look into "https://s3.amazonaws.com/dd-agent-mstesting/builds/stable/installers_v2.json" for the Datadog Installer
+// version "7.56.0-installer-0.4.5-1"
+func WithInstallerURLFromInstallersJSON(bucket, channel, version string) InstallerOption {
+	return func(params *InstallerParams) error {
+		url, err := installers.GetProductURL(fmt.Sprintf("https://s3.amazonaws.com/%s/builds/%s/installers_v2.json", bucket, channel), "datadog-installer", version, "x86_64")
+		if err != nil {
+			return err
+		}
+		params.installerURL = url
+		return nil
+	}
+}
+
+// Install will attempt to install the Datadog Installer on the remote host.
+// By default, it will use the installer from the current pipeline.
+func (d *DatadogInstaller) Install(opts ...InstallerOption) error {
+	params := InstallerParams{}
+	err := optional.ApplyOptions(&params, opts)
+	if err != nil {
+		return nil
+	}
+	if params.installerURL == "" {
+		artifactURL, err := pipeline.GetPipelineArtifact(d.env.AwsEnvironment.PipelineID(), pipeline.AgentS3BucketTesting, pipeline.DefaultMajorVersion, func(artifact string) bool {
+			return strings.Contains(artifact, "datadog-installer")
```

Comment:
check that the extension is `.msi`

If we don't save the debug symbols yet we should add that, and checking the file extension now will prevent this from breaking when we get to that.

ditto for the component